web/src/components/navBar/Nav.tsx

```diff
@@ -0,0 +1,258 @@
+import { ReactElement } from "react-markdown/lib/react-markdown";
+import { NavLink as RouterLink } from "react-router-dom";
+
+import SingleStoreLogoDrak from "@/assets/singlestore-logo-dark.svg";
+import SinglestoreLogo from "@/assets/singlestore-logo-filled-sm.svg";
+import { GithubStargazer } from "@/components/GithubButtons";
+
+export const Nav = () => {
+  const { toggleColorMode } = useColorMode();
+  const { colorMode } = useColorMode();
+  const navMenu = useDisclosure();
+  const [isSmallScreen] = useMediaQuery("(max-width: 640px)");
+  let themeModeIcon = <MoonIcon cursor="pointer" onClick={toggleColorMode} />;
+
+  const NavLinkActiveButtonStyle = {
+    background: useColorModeValue("#4F34C7", "#CCC3F9"),
+    color: useColorModeValue("#FFFFFF", "#2F206E"),
+  };
+
+  const LinksIntenralComponent = ({
+    NavLinkTitle,
+    IconElement,
+  }: {
+    NavLinkTitle: string;
+    IconElement: ReactElement;
+  }) => {
+    return (
+      <Flex gap={2} alignItems="center">
+        <b>{NavLinkTitle} </b>
+        <Stack fontSize="0.8em" alignItems="center">
+          {IconElement}
+        </Stack>
+      </Flex>
+    );
+  };
+
+  const NavLinkComponent = ({
+    NavLinkTitle,
+    IconElement,
+    to,
+  }: {
+    NavLinkTitle: string;
+    IconElement: ReactElement;
+    to: string;
+  }) => {
+    return (
+      <Link
+        padding="4px 15px 4px 15px"
+        borderRadius="5px"
+        as={RouterLink}
+        color={useColorModeValue("#553ACF", "#CCC3F9")}
+        to={to}
+        onClick={navMenu.onClose}
+        _hover={NavLinkActiveButtonStyle}
+        _activeLink={NavLinkActiveButtonStyle}
+      >
+        <LinksIntenralComponent
+          NavLinkTitle={NavLinkTitle}
+          IconElement={IconElement}
+        />
+      </Link>
+    );
+  };
+
+  const links = (
+    <>
+      <NavLinkComponent
+        to="/dashboard"
+        NavLinkTitle="Dashboard"
+        IconElement={<Icon as={colorMode === "light" ? BsMap : BsMapFill} />}
+      />
+      <NavLinkComponent
+        to="/analytics"
+        NavLinkTitle="Analytics"
+        IconElement={
+          <Icon
+            border="1px"
+            padding="1px"
+            borderRadius="2px"
+            as={colorMode === "light" ? BsBarChart : BsFillBarChartFill}
+          />
+        }
+      />
+      <NavLinkComponent
+        to="/configure"
+        NavLinkTitle="Configure"
+        IconElement={<Icon as={colorMode === "light" ? BsGear : BsGearFill} />}
+      />
+    </>
+  );
+
+  const handleLinkRedirects = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => {
+    window.open(e.currentTarget.value, "_blank");
+  };
+
+  const handleHamburgerNavMenu = () => {
+    if (navMenu.isOpen) {
+      return (
+        <Box pb={4} display={{ md: "none" }}>
+          <Stack as="nav" spacing={4}>
+            {links}
+          </Stack>
+        </Box>
+      );
+    }
+    return null;
+  };
+
+  if (colorMode === "dark") {
```

Comment:
nit: Have the variable initialization next to the if that can change it. Makes it easier to understand what's being done here.

```suggestion
  let themeModeIcon = <MoonIcon cursor="pointer" onClick={toggleColorMode} />;
  if (colorMode === "dark") {
```

Reply:
have made his changes